lib/javascript/JavascriptModulesPlugin.js

```diff
@@ -839,18 +839,25 @@ class JavascriptModulesPlugin {
 				startupSource.add(`var ${RuntimeGlobals.exports} = {};\n`);
 			}
 
-			const renamedInlinedModule = this.renameInlineModule(
-				allModules,
-				renderContext,
-				inlinedModules,
-				chunkRenderContext,
-				hooks
-			);
+			const entryIife = compilation.options.optimization.entryIife;
+			/** @type {Map<Module, Source> | false} */
+			let renamedInlinedModule = false;
+			if (entryIife) {
+				renamedInlinedModule = this.getRenamedInlineModule(
+					allModules,
+					renderContext,
+					inlinedModules,
+					chunkRenderContext,
+					hooks,
+					allStrict,
+					Boolean(chunkModules)
+				);
+			}
 
 			for (const m of inlinedModules) {
-				const renderedModule =
-					renamedInlinedModule.get(m) ||
-					this.renderModule(m, chunkRenderContext, hooks, false);
+				const renderedModule = renamedInlinedModule
+					? renamedInlinedModule.get(m)
```

Comment:
We need a test here, I think it is unnecessary, because we already rename them before

Reply:
Just try
```js
const renderedModule = this.renderModule(
					m,
					chunkRenderContext,
					hooks,
					false
				);
```

And the output the same

Reply:
> We need a test here, I think it is unnecessary, because we already renamed them before

It's necessary. The initial value of `⁠renamedInlinedModule` is `false`. If `⁠avoidEntryIife` is enabled, ⁠`getRenamedInlineModule` will check for other reasons to bail out of the IIFE. If any exist, `⁠renamedInlinedModule` will remain false because the inlined module will still be wrapped in an IIFE for those reasons. Therefore, we skip the optimization here.

Reply:
I ask this because of https://github.com/webpack/webpack/blob/fe169983cb41f2a4d789468c7bd5cfbe860919de/lib/javascript/JavascriptModulesPlugin.js#L1514

Do we really need `source.replace`? We need a test here, because our tests don't need renaming, so i thought we could avoid this here

Reply:
If we don't need renaming, we can simplify our code quite a bit.

Reply:
> Do we really need source.replace? 

Yes, to avoid name collision.

> We need a test here, because our tests don't need renaming, so i thought we could avoid this here

We already have, check `test/configCases/output-module/iife-entry-module-with-others/test.js`, when `avoidEntryIife` is enabled, we remove the IIFE by renaming. I'm not sure if I'm answering the question you had in mind. 😂

Reply:
@fi3ework I still think we have wrong logic, because if I remove
```js
for (const variable of usedInNonInlined) {
				const references = getAllReferences(variable);
				const allIdentifiers = new Set(
					references.map(r => r.identifier).concat(variable.identifiers)
				);

				const newName = this.findNewName(
					variable.name,
					usedNames,
					m.readableIdentifier(runtimeTemplate.requestShortener)
				);
				usedNames.add(newName);
				for (const identifier of allIdentifiers) {
					const r = /** @type {Range} */ (identifier.range);
					const path = getPathInAst(ast, identifier);
					if (path && path.length > 1) {
						const maybeProperty =
							path[1].type === "AssignmentPattern" && path[1].left === path[0]
								? path[2]
								: path[1];
						if (maybeProperty.type === "Property" && maybeProperty.shorthand) {
							source.insert(r[1], `: ${newName}`);
							continue;
						}
					}
					source.replace(r[0], r[1] - 1, newName);
				}
			}
```

Everything works fine

Reply:
As I said before, we already doing renaming before (in concatination modules) and I don't think we need to do it here

Reply:
Just put `console.log` or use `debugger` for fast debug and we never reach this branches

We already continue here:
```
if (usedInNonInlined.size === 0) {
  renamedInlinedModules.set(m, source);
  continue;
}
```

Reply:
The case in `test/configCases/output-module/iife-entry-module-with-others/index.js` It's not working as well as it should. I'm fixing it. I checked the 'test/cases/entry-inline' case and it is still working.

I made https://stackblitz.com/edit/node-ojwwax, you could check `localVar` in `dist-5.91.0` and `dist-5.94.0` to see the difference. In 5.94.0, `localVar` is renamed `src_localVar` by renaming.

Reply:
@alexander-akait  Updated, please check the new test case now. 
The current test cases serve the following purposes, for instance with `iife-entry-module-with-others`:

1. Assert the correct IIFE comment in the bundle.
2. Assert the correct runtime behavior of the bundle.
3. If the reason branch `it need to be isolated against other entry modules.` is deleted, the test will fail 

The other test cases are similar, `iife-multiple-entry-modules` has an extra `avoidEntryIife` configuration toggle test, which can be seen as a supplement to `test/cases/entry-inline`.